### PR TITLE
auth/tests: unmock ClientCall

### PR DESCRIPTION
**Pull request by carl-mastrangelo**

### CODE REVIEW

**Comment by ejona86** [comment]
@carl-mastrangelo LGTM. The class under test is also on the chopping block, once we are able to migrate enough people away.

**Comment by carl-mastrangelo** [comment]
@ejona86 PTAL.  I added the rest of the classes where it was being mocked.  There's a vague todo in my mind to normalize the dummy Calls, but right now I don't feel like making a place where they can all be shared.

File: stub/build.gradle [hunk 1]
```diff
@@ -5,6 +5,7 @@ plugins {
 description = "gRPC: Stub"
 dependencies {
     compile project(':grpc-core')
+    testCompile project(':grpc-testing')
```
Comment (ejona86) [inline stub/build.gradle]: Why is this necessary?

Comment (carl-mastrangelo) [inline stub/build.gradle]: For Truth.assertThat

Comment (ejona86) [inline stub/build.gradle]: ?? But that is part of `libraries.truth`. Why the need for grpc-testing?

Comment (carl-mastrangelo) [inline stub/build.gradle]: Hmm, I suppose there isn't.  I think this is how core pulled it in.  I'll switch it.

**Comment by ejona86** [comment]
LGTM
